Signal/src/AppDelegate.m

```diff
@@ -239,7 +239,11 @@ - (void)applicationDidBecomeActive:(UIApplication *)application {
                                                [TSSocketManager becomeActiveFromForeground];
                                                [[Environment getCurrent].contactsManager verifyABPermission];
                                            }];
-
+    
+    if ([TSAccountManager isRegistered]) {
+        [[PushManager sharedManager] applicationDidBecomeActive];
```

Comment:
I see no problem with dispatching message fetching off the main thread.

If your concern was with thread safety at the storage level specifically, I don't see a problem there either. But it would be good to dive into the concurrency idioms. They're different: https://github.com/yapstudios/YapDatabase/wiki/Thread-Safety